public/app/features/alerting/unified/components/contact-points/useNotificationTemplates.ts

```diff
@@ -0,0 +1,138 @@
+import { produce } from 'immer';
+
+import { ActiveTab as ContactPointsActiveTabs } from 'app/features/alerting/unified/components/contact-points/ContactPoints';
+import { useDispatch } from 'app/types';
+
+import { AlertManagerCortexConfig } from '../../../../../plugins/datasource/alertmanager/types';
+import { alertmanagerApi } from '../../api/alertmanagerApi';
+import { updateAlertManagerConfigAction } from '../../state/actions';
+import { ensureDefine } from '../../utils/templates';
+import { TemplateFormValues } from '../receivers/TemplateForm';
+
+// TODO Check the issue with useAlertmanagerConfig hook
+export function useNotificationTemplates(amSourceName?: string) {
+  const templatesRequestState = alertmanagerApi.endpoints.getAlertmanagerConfiguration.useQuery(amSourceName || '', {
+    skip: !amSourceName,
+    selectFromResult: (state) => ({
+      ...state,
+      data: state.data ? amConfigToTemplates(state.data) : undefined,
+      currentData: state.currentData ? amConfigToTemplates(state.currentData) : undefined,
+    }),
+  });
+
+  return templatesRequestState;
+}
+
+export interface NotificationTemplate {
+  name: string;
+  template: string;
+  provenance?: string;
+}
+
+function amConfigToTemplates(config: AlertManagerCortexConfig): NotificationTemplate[] {
+  return Object.entries(config.template_files).map(([name, template]) => ({
+    name,
+    template,
+    provenance: (config.template_file_provenances ?? {})[name],
+  }));
+}
+
+export function useCreateNotificationTemplate(amSourceName?: string) {
+  const dispatch = useDispatch();
+  const [fetchAmConfig] = alertmanagerApi.endpoints.getAlertmanagerConfiguration.useLazyQuery();
+
+  return async (templateValues: TemplateFormValues) => {
+    if (!amSourceName) {
+      return Promise.reject(new Error('Alertmanager source name is required.'));
+    }
```

Comment:
I think it's best to assume that we definitely have the alertmanager name outside of the hooks, and we worry about the falsy case there?